src/audio/rtnr/rtnr.c

```diff
@@ -399,26 +413,66 @@ static int rtnr_cmd_get_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int rtnr_set_bin_data(struct comp_dev *dev,
+				       struct sof_ipc_ctrl_data *cdata)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	assert(cd);
+
+	comp_dbg(dev, "rtnr_set_bin_data() msg_index = %d, num_elems = %d, remaining = %d ",
+		 cdata->msg_index, cdata->num_elems,
+		 cdata->elems_remaining);
+
+	switch (cdata->data->type) {
+	case SOF_RTNR_CONFIG:
+		comp_dbg(dev, "rtnr_set_bin_data(): SOF_RTNR_CONFIG");
+		ret = comp_data_blob_set_cmd(cd->model_handler, cdata);
+		if (ret >= 0)
+			ret = rtnr_check_config_validity(dev, cd);
+		break;
+	case SOF_RTNR_DATA_1:
+		comp_dbg(dev, "rtnr_set_bin_data(): SOF_RTNR_DATA_1");
+		ret = RTKMA_API_Set_Data_1(cd->rtk_agl,
+									cdata->data->data,
+									cdata->msg_index,
+									cdata->num_elems,
+									cdata->elems_remaining);
+		break;
+	case SOF_RTNR_DATA_2:
+		comp_dbg(dev, "rtnr_set_bin_data(): SOF_RTNR_DATA_2");
+		ret = RTKMA_API_Set_Data_2(cd->rtk_agl,
```

Comment:
Yes it could be a large blob. As far as I know the blob handler will allocate a buffer to hold all the incoming data, then the component retrieves the buffer later. If the blob is too large, than the buffer allocation might fail. We want to modify the data in RTNR directly. Thus we choose not not use blob handler.

Reply:
got it, please add a comment explaining that.

Reply:
The commit message is updated. Thanks!